File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int l =0;
        int r =0;
        int  p =0;
        for(int i = 0;i<prices.size();i++){
            if(prices[l] > prices[i]){
                l = i;
                r = i;
            }
            else if(prices[r] < prices[i]){
                r = i;
                p = prices[r] - prices[l];
                profit = max(p,profit); 
            }
            
        }
        return profit;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l = 1
        r = max(piles)
        while l<=r:
            m = (l+r)//2
            cnt = sum(math.ceil(p/m) for p in piles)
            if cnt <= h:
                r = m - 1
            else:
                l = m + 1
        return l